chrome/content/coniks/coniks.js

```diff
@@ -0,0 +1,101 @@
+this.EXPORTED_SYMBOLS = ["coniks"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("resource:///modules/imXPCOMUtils.jsm");
+Cu.import("resource:///modules/jsProtoHelper.jsm");
+Cu.import("chrome://otr/content/coniks/context.js");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/otr.js");
+
+function accountname(acc) {
+  let account = acc.normalizedName;
+  let protocol = acc.protocol.normalizedName;
+  return `${account}@${protocol}`;
+}
+
+var coniks = {
+  isEnable: false,
+
+  init: function() {
+    coniks.setPrefs();
+    coniks.isEnabled = coniks.prefs.getBoolPref("enabled");
+    if (!coniks.isEnabled)
+      return;
+
+    this._contexts = new Map();
+    for (let acc of helpers.getAccounts()) {
+      coniks._contexts.set(accountname(acc), new AccountContext({
+        account: acc.normalizedName,
+        protocol: acc.protocol.normalizedName
+      }));
+    }
+
+    coniks.loadFiles();
+    Services.obs.addObserver(coniks, "account-added", false);
+  },
+
+  observe: function(aObject, aTopic, aMsg) {
+    switch(aTopic) {
+    case "account-added":
+      coniks.onAccountCreated(aObject);
```

Comment:
There's a race condition here, because we also generate a key on "account-added".

Maybe we should remove this observer and just call `coniks.onAccountCreated` from the event handler in ui after a successful key generation.